lib/introspective_grape/utils/json_expander.rb

```diff
@@ -0,0 +1,21 @@
+module IntrospectiveGrape
+  module Utils
+    module JsonExpander
+      extend self
+
+      # Expands an object with its #to_json method if it's not a primitive or if the first element is a Grape::Entity.
+      def expand(object)
+        if (object.respond_to?(:to_json) &&
+          [String, Symbol, Hash, Array, NilClass, TrueClass, FalseClass, Numeric].all? { |x| !object.is_a?(x) }) ||
+          # Force arrays of Grape::Entities into their hash representations before camelizing
+          (object.is_a?(Array) && defined?(Grape::Entity) && object.first.is_a?(Grape::Entity))
+
```

Comment:
I remember the point of the increasingly complicated logic here was to not do `JSON.parse(object.to_json)` if the object was already a hash, or an array of hashes. Maybe something like

```
o = object.kind_of?(Array) ? object.first : object 
if o.respond_to?(:to_json) && !o.respond_to?(:with_camel_keys)
  JSON.parse(object.to_json)
else
  object
end
```

Would be simpler? I'm not a huge fan of my kludge for the array situation here expecting the array elements to all be the same as the first and only dealing with the situation for Grape::Entity, since it would't work if the gem were to someday support other facade classes...